File: dms-frontend-documents/src/app/model/file-element.ts
export class FileElement {
  id: string;
  isFolder: boolean;
  name: string;
  parent: string;
  children?: Array<FileElement>;

  constructor(id: string, isFolder: boolean, name: string, parent: string) {
    this.id = id;
    this.isFolder = isFolder;
    this.name = name;
    this.parent = parent;
    if (isFolder) {
      this.children = new Array<FileElement>();
    }
  }
}

File: dms-backend-documents/src/main/java/com/dms/backend/documents/model/FileElement.java
package com.dms.backend.documents.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;

import java.io.Serializable;
import java.util.List;

@Data
public class FileElement implements Serializable {
    private String name;
    private boolean isFolder;
    private String path;
    private List<FileElement> children;
    @JsonIgnore
    private byte[] file;

    @Override
    public String toString() {
        return "FileElement{" +
                "name='" + name + '\'' +
                ", isFolder=" + isFolder +
                ", path='" + path + '\'' +
                ", children=" + children +
                '}';
    }
}
